src/test_wasm_simd.py

```diff
@@ -0,0 +1,127 @@
+#! /usr/bin/env python
+
+import os
+import re
+import subprocess
+import sys
+import tempfile
+
+wasm_simd_test_cases = [
+  # Oper Arity ArgType     ReturnType   Opcode   KnownFailure
+  ('add',  2, 'int32x4',   'int32x4',   'add',   False),
+  ('add',  2, 'int16x8',   'int16x8',   'add',   False),
+  ('add',  2, 'int8x16',   'int8x16',   'add',   False),
+  ('sub',  2, 'int32x4',   'int32x4',   'sub',   False),
+  ('sub',  2, 'int16x8',   'int16x8',   'sub',   False),
+  ('sub',  2, 'int8x16',   'int8x16',   'sub',   False),
+  ('mul',  2, 'int32x4',   'int32x4',   'mul',   False),
+  ('mul',  2, 'int16x8',   'int16x8',   'mul',   False),
+  ('mul',  2, 'int8x16',   'int8x16',   'mul',   False),
+  ('add',  2, 'float64x2', 'float64x2', 'fadd',  False),
+  ('add',  2, 'float32x4', 'float32x4', 'fadd',  False),
+  ('sub',  2, 'float64x2', 'float64x2', 'fsub',  False),
+  ('sub',  2, 'float32x4', 'float32x4', 'fsub',  False),
+  ('mul',  2, 'float64x2', 'float64x2', 'fmul',  False),
+  ('mul',  2, 'float32x4', 'float32x4', 'fmul',  False),
+  ('div',  2, 'float64x2', 'float64x2', 'fdiv',  False),
+  ('div',  2, 'float32x4', 'float32x4', 'fdiv',  False),
+  ('min',  2, 'float64x2', 'float64x2', 'fmin',  True ),
+  ('min',  2, 'float32x4', 'float32x4', 'fmin',  True ),
+  ('max',  2, 'float64x2', 'float64x2', 'fmax',  True ),
+  ('max',  2, 'float32x4', 'float32x4', 'fmax',  True ),
+  ('abs',  1, 'float64x2', 'float64x2', 'fabs',  True ),
+  ('abs',  1, 'float32x4', 'float32x4', 'fabs',  True ),
+  ('neg',  1, 'float64x2', 'float64x2', 'fsub',  False),
+  ('neg',  1, 'float32x4', 'float32x4', 'fsub',  False),
+  ('sqrt', 1, 'float64x2', 'float64x2', 'fsqrt', True ),
+  ('sqrt', 1, 'float32x4', 'float32x4', 'fsqrt', True ),
+]
+
+def create_test_file(datatype, operation, arity, returntype):
+  f = tempfile.NamedTemporaryFile(mode='w', suffix='.c', delete=False)
+  f.write('#include <emscripten/vector.h>\n\n')
+  if arity == 2:
+    f.write('%s test(%s a, %s b) {\n' % (returntype, datatype, datatype))
+    f.write('  return emscripten_%s_%s(a, b);\n' % (datatype, operation))
+    f.write('}\n')
+  else:
+    f.write('%s test(%s a) {\n' % (returntype, datatype))
+    f.write('  return emscripten_%s_%s(a);\n' % (datatype, operation))
+    f.write('}\n')
+  f.close()
+  return f.name
+
+def compile_test_file(filename, clang, include):
+  p = subprocess.Popen([clang, '-w', '-S', '-emit-llvm', filename,
+                        '-O3', '-o-', '-I', include],
+                       stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  (output, err) = p.communicate()
```

Comment:
Call these "stderr, stdout" perhaps for consistency?

Also I think that the python style is to not include braces when destructuring like this.